File: app/src/main/java/io/github/marcinrogacki/goworld/Character.java
package io.github.marcinrogacki.goworld;

import java.util.Random;

public class Character {
    private int life = 100;
    private int damage = 10;

    public void attack(Character subject)
    {
        subject.takeDamage(getDamage());
    }

    public void takeDamage(int amount)
    {
        life = life - amount;
        if (life < 0) {
            life = 0;
        }
    }

    public void heal()
    {
        life = life + getDamage() * 2;
        if (life > 100) {
            life = 100;
        }
    }

    public int getLife()
    {
        return life;
    }

    public int getDamage()
    {
        Random rand = new Random();
        return rand.nextInt(10);
    }

    public String getLifeText()
    {
        if (life > 80) {
            return "Healthy";
        } else if (life > 60) {
            return "Injured";
        } else if (life > 40) {
            return "Wounded";
        } else if (life > 20) {
            return "Damaged";
        } else if (life > 0) {
            return "Close to dead";
        } else {
            return "You are dead";
        }
    }
}

File: app/src/main/java/io/github/marcinrogacki/goworld/Character.kt
package io.github.marcinrogacki.goworld

import java.util.*

class Character {
    var life = 100
        private set

    private val damage: Int
        get() {
            // Range: 0-9
            return Random().nextInt(10)
        }

    private val aid: Int
        get() {
            // Range: 2-6
            return Random().nextInt(4) + 2
        }

    fun attack(subject: Character) {
        subject.takeDamage(damage)
    }

    private fun takeDamage(amount: Int) {
        life -= amount
        if (life < 0) {
            life = 0
        }
    }

    fun heal() {
        life += aid
        if (life > 100) {
            life = 100
        }
    }

    val lifeText: String
        get() = when {
            life > 80 -> {
                "Healthy"
            }
            life > 60 -> {
                "Injured"
            }
            life > 40 -> {
                "Wounded"
            }
            life > 20 -> {
                "Damaged"
            }
            life > 0 -> {
                "Fatal"
            }
            else -> {
                "Dead"
            }
        }
}
